codegen/src/generator/rest_xml.rs

```diff
@@ -0,0 +1,855 @@
+    if operation.input.is_none() {
+        return None;
+    }
+
+    let shape = service.shapes.get(&operation.input.as_ref().unwrap().shape).unwrap();
+
+    Some(shape.members.as_ref().unwrap().iter().filter_map(|(member_name, member)| {
+        if member.location.is_none() {
+            return None;
+        }
+        match &member.location.as_ref().unwrap()[..] {
+            "uri" => {
+                if shape.required(&member_name) {
+                    Some(format!("request_uri = request_uri.replace(\"{{{location_name}}}\", &input.{field_name});",
+                        location_name = member.location_name.as_ref().unwrap(),
+                        field_name = member_name.to_snake_case()))
+                } else {
+                    Some(format!("request_uri = request_uri.replace(\"{{{location_name}}}\", &input.{field_name}.unwrap());",
+                        location_name = member.location_name.as_ref().unwrap(),
+                        field_name = member_name.to_snake_case()))
+                }
+            },
+            _ => None
+        }
+    }).collect::<Vec<String>>().join("\n"))
+}
+
+fn generate_headers(service: &Service, operation: &Operation) -> Option<String> {
+
+    // nothing to do if there's no input type
+    if operation.input.is_none() {
+        return None;
+    }
+
+    let shape = service.shapes.get(&operation.input.as_ref().unwrap().shape).unwrap();
+
+    Some(shape.members.as_ref().unwrap().iter().filter_map(|(member_name, member)| {
+        if member.location.is_none() {
+            return None;
+        }
+        match &member.location.as_ref().unwrap()[..] {
+            "header" => {
+                if shape.required(&member_name) {
+                    Some(format!("request.add_header(\"{location_name}\", &input.{field_name});",
+                        location_name = member.location_name.as_ref().unwrap(),
+                        field_name = member_name.to_snake_case()))
+                } else {
+                    Some(format!("
+                        if let Some(ref {field_name}) = input.{field_name} {{
+                            request.add_header(\"{location_name}\", &{field_name}.to_string());
+                        }}",
+                        location_name = member.location_name.as_ref().unwrap(),
+                        field_name = member_name.to_snake_case()))
+                }
+            },
+            _ => None
+        }
+    }).collect::<Vec<String>>().join("\n"))
+}
+
+fn generate_parameters(service: &Service, operation: &Operation) -> Option<String> {
+
+    // nothing to do if there's no input type
+    if operation.input.is_none() {
+        return None;
+    }
+
+    let shape = service.shapes.get(&operation.input.as_ref().unwrap().shape).unwrap();
+
+    Some(shape.members.as_ref().unwrap().iter().filter_map(|(member_name, member)| {
+        if member.location.is_none() {
+            return None;
+        }
```

Comment:
Ooh, match guards.  There's a bit of new syntax for me to file away for future use.  `Some(location) if location == "querystring"`